### ISSUE TITLE
Snapshot test - do we really need it?

### ISSUE THREAD

**Issue by nolanlawson** [Level/abstract-leveldown#38]
I noticed [this test](https://github.com/rvagg/abstract-leveldown/blob/8832454ee2150a157072c398b9636967e9e2e80a/abstract/iterator-test.js#L458-L480) was added recently. It's causing some headaches for me in localstorage-down, but I'm wondering how many other *down authors have actually implemented this thing?

Reading the test, I can understand the problem it's trying to solve. Ideally users should be able to open up a read-only iterator against a database and continue to read from a "snapshot" even as others are writing to it.

However, that's a really, really involved feature (concurrency! transactions!), especially for simple modules like MemDOWN and localstorage-down. And this test doesn't even seem to be very thorough. I could adhere to the letter of the law by just passing this one test, while still not really implementing the proper transactional semantics.

What's the feeling about this? Should *down authors pick-and-choose the tests we know we won't support, or should we rise to the occasion and try to pass tests like this?

**Comment by timkuijsten** [comment]
While implementing IndexedDB for Level.js I've encountered the problem that, although idb cursors support snapshotting, browsers time them out aggressively, meaning that if _next isn't called within the next tick, there is a high probability that the cursor timed out. I've added a `reopenOnTimeout` option to the iterator, but this defeats the snapshot property. Also, the current set of abstract tests are not "fast enough" to prevent timeouts, as noted in https://github.com/maxogden/level.js/issues/46#issuecomment-183709506. I'm wondering if I should change the default of reopenOnTimeout to true, a.k.a. non-snapshot mode. Then all tests pass (even the abstract snapshot tests since they are fast enough) but this differs from the LevelDB defaults and what a user might expect.
